src/WPF/ArcGISRuntime.WPF.Viewer/Samples/Network Analysis/FindServiceArea/FindServiceArea.xaml.cs

```diff
@@ -0,0 +1,264 @@
+
+        }
+        private async void ShowServiceAreas_Click(object sender, System.Windows.RoutedEventArgs e)
+        {
+            // Enable both facility and barrier buttons.
+            AddBarrier.IsEnabled = true;
+            AddFacilities.IsEnabled = true;
+
+            // Turn off any drawing mode.
+            _facilityMode = false;
+            _barrierMode = false;
+
+            await CreateServiceArea();
+
+            // Check that there is at least 1 facility to find a service area for.
+            if (_serviceAreaFacilities.Count > 0)
+            {
+                List<PolylineBarrier> polylineBarriers = new List<PolylineBarrier>();
+                foreach (Graphic barrier in _barrierOverlay.Graphics.ToList())
+                {
+                    polylineBarriers.Add(new PolylineBarrier((Polyline) barrier.Geometry));
+                }
+
+                // Add the barriers to the service area parameters.
+                _serviceAreaParameters.SetPolylineBarriers(polylineBarriers);
+
+                // Clear existing graphics for service areas.
+                _serviceAreasOverlay.Graphics.Clear();
+
+                // Update the parameters to include all of the placed facilities.
+                _serviceAreaParameters.SetFacilities(_serviceAreaFacilities);
+
+                try
+                {
+                    ServiceAreaResult result = await _serviceAreaTask.SolveServiceAreaAsync(_serviceAreaParameters);
+
+                    // Loop over each facility.
+                    for (int i = 0; i < _serviceAreaFacilities.Count; i++)
+                    {
+                        // Create list of polygons from a service facility.
+                        List<ServiceAreaPolygon> polygons = result.GetResultPolygons(i).ToList();
+
+                        // Loop over every polygon in every facilities result.
+                        for (int j = 0; j<polygons.Count; j++)
+                        {
+                            // Add graphic for service area. Alternate the color of each polygon.
+                            _serviceAreasOverlay.Graphics.Add(new Graphic(polygons[j].Geometry, _fillSymbols[j%2]));
+                        }
+                    }
+                }
+                catch (Esri.ArcGISRuntime.Http.ArcGISWebException exception)
+                {
+                    if(exception.Message.ToString().Equals("Unable to complete operation."))
+                    {
+                        System.Windows.MessageBox.Show("Facility not within San Diego area!", "Sample error");
+                    }
+                    else
+                    {
+                        System.Windows.MessageBox.Show("An ArcGIS web exception occurred. \n" + exception.Message.ToString(), "Sample error");
+                    }
+                }
+            }
+            else
+            {
+                System.Windows.MessageBox.Show("Must have at least 1 Facility!", "Sample error");
```

Comment:
I think it would be more readable to flip this branch:
```csharp
// Check that there is at least 1 facility to find a service area for.
if (_serviceAreaFacilities.Count > 0)
{
    System.Windows.MessageBox.Show("Must have at least 1 Facility!", "Sample error");
    return;
}
```